### ISSUE TITLE
Force deep sleep regardless of WiFi successful connection.

### ISSUE THREAD

**Issue by mr-sneezy** [esphome/feature-requests#210]
**Describe the problem you have/What new integration you would like**

Additional Deep Sleep parameter if it's actually possible.
Go into Deep Sleep regardless of a successful WiFi connection or not.
Currently the behaviour as far as I can see is to go into deep sleep after the time out only AFTER a successful Wifi connect to HA.

**Please describe your use case for this integration and alternatives you've tried:**

Project is a vehicle battery monitor. 
When parked at home the ESP8266 circuit reads the sensors (including ADC) and sleeps for 60 mins. Current drops to a few mA.
When away from home the ESP8266 keeps trying to connect unsuccessfully and the deep sleep does not occur (as long as I waited) and current is 80mA.

**Additional context**

While a car battery is no issue for a day or two, a scooter battery is a problem when parked away from home with circuit connected.
I see there are other similar requests but not quite the same in regards to an option to force a deep sleep without WiFi connecting first...
